src/test/java/xades4j/production/SignerSpecificTest.java

```diff
@@ -0,0 +1,130 @@
+package xades4j.production;
+
+import org.bouncycastle.asn1.ASN1Encodable;
+import org.bouncycastle.asn1.DERBMPString;
+import org.bouncycastle.asn1.DERUTF8String;
+
+
+import org.bouncycastle.asn1.x500.AttributeTypeAndValue;
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.asn1.x500.X500NameBuilder;
+
+import org.bouncycastle.asn1.x500.style.RFC4519Style;
+
+import org.bouncycastle.asn1.x509.BasicConstraints;
+import org.bouncycastle.asn1.x509.Extension;
+import org.bouncycastle.asn1.x509.KeyUsage;
+import org.bouncycastle.cert.X509v3CertificateBuilder;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
+import org.bouncycastle.cert.jcajce.JcaX509v3CertificateBuilder;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder;
+import org.junit.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import xades4j.algorithms.EnvelopedSignatureTransform;
+import xades4j.properties.DataObjectDesc;
+
+import xades4j.providers.impl.DirectKeyingDataProvider;
+
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.math.BigInteger;
+
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.SecureRandom;
+import java.security.Security;
+import java.security.cert.X509Certificate;
+import java.util.Date;
+import java.util.Random;
+import java.util.ResourceBundle;
+
+
+/**
+ * @author Artem R. Romanenko
+ * @version 02/04/2018
+ */
+public class SignerSpecificTest extends SignerBESTest {
+    private final static ResourceBundle BUNGLE = ResourceBundle.getBundle(SignerSpecificTest.class.getName());
+
+    @Test
+    public void signWithNationalCertificate() throws Exception {
```

Comment:
If I understood correctly, the purpose is testing X500 names with cyrillic. Maybe the method could be named accordingly?

Reply:
Any national alphabet. I added arabic for example.